dbfopen.c

```diff
@@ -1842,18 +1949,58 @@ int SHPAPI_CALL DBFDeleteField(DBFHandle psDBF, int iField)
     /* resize fields arrays */
     psDBF->nFields--;
 
-    psDBF->panFieldOffset = STATIC_CAST(
+    psDBF_realloc_ptr = STATIC_CAST(
```

Comment:
Actually reallocating when removing a field is totally useless. You should just remove the reallocs